clippy_lints/src/methods/mod.rs

```diff
@@ -3261,6 +3262,41 @@ declare_clippy_lint! {
     "calling `drain` in order to `clear` a container"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for initial `'/ or \\'` in an argument to `.join()` on a `Path`.
+    ///
+    /// ### Why is this bad?
+    /// `.join()` comments starting with a separator  (`/` or `\\`) can replace the entire path.
+    /// If this is intentional, prefer creating a new `Path` instead.
+    ///
+    /// See [`Path::join()`](https://doc.rust-lang.org/std/path/struct.Path.html#method.join)
+    ///
+    /// ### Example
+    /// ```rust
+    /// let path = std::path::Path::new("/bin");
+    /// let res = path.join("/sh");
+    /// assert_eq!(res, std::path::PathBuf::from("/sh"));
```

Comment:
In these examples, it would be nicer to use `# use std::path::Path` and then remove the leading `std::path`.

```suggestion
    /// # use std::path::{Path, PathBuf}
    /// let path = Path::new("/bin");
    /// let joined_path = path.join("/sh");
    /// assert_eq!(joined_path, PathBuf::from("/sh"));
```

Same in the example below